sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/aio/_client.py

```diff
@@ -176,8 +179,14 @@ class KeyClient(AsyncKeyVaultClientBase):
                 :caption: Delete a key
                 :dedent: 8
         """
-        bundle = await self._client.delete_key(self.vault_endpoint, name, error_map=_error_map, **kwargs)
-        return DeletedKey._from_deleted_key_bundle(bundle)
+        polling_interval = kwargs.pop("_polling_interval", 2)
+        deleted_key = DeletedKey._from_deleted_key_bundle(
+            await self._client.delete_key(self.vault_endpoint, name, error_map=_error_map, **kwargs)
+        )
+        command = partial(self.get_deleted_key, name=name, **kwargs)
+
+        delete_key_poller = DeleteResourcePollerAsync(interval=polling_interval)
+        return await async_poller(command, deleted_key, None, delete_key_poller)
```

Comment:
If we didn't await this call wouldn't we get the behavior we want in the sync case as well? Seems like without this await the calling code could be like this:

~~~python
delete_op = await begin_delete_key('mykey')
await delete_op.wait()
~~~

Reply:
i think the async poller is just going to follow the guidelines for now and will be changed later, thanks!